pkg/operator/controller/controller.go

```diff
@@ -172,30 +174,62 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 // enforceEffectiveIngressDomain determines the effective ingress domain for the
 // given clusteringress and ingress configuration and publishes it to the
 // clusteringress's status.
-func (r *reconciler) enforceEffectiveIngressDomain(ci *operatorv1.IngressController, ingressConfig *configv1.Ingress) error {
+func (r *reconciler) enforceEffectiveIngressDomain(ic *operatorv1.IngressController, ingressConfig *configv1.Ingress) error {
 	// The clusteringress's ingress domain is immutable, so if we have
 	// published a domain to status, we must continue using it.
-	if len(ci.Status.Domain) > 0 {
+	if len(ic.Status.Domain) > 0 {
 		return nil
 	}
 
-	updated := ci.DeepCopy()
+	updated := ic.DeepCopy()
+	var domain string
 	switch {
-	case len(ci.Spec.Domain) > 0:
-		updated.Status.Domain = ci.Spec.Domain
+	case len(ic.Spec.Domain) > 0:
+		domain = ic.Spec.Domain
 	default:
-		updated.Status.Domain = ingressConfig.Spec.Domain
+		domain = ingressConfig.Spec.Domain
+	}
+	unique, err := r.isDomainUnique(domain)
+	if err != nil {
+		return err
+	}
+	if !unique {
+		log.Info("domain not unique, not setting status domain for IngressController", "namespace", ic.Namespace, "name", ic.Name)
+		return nil
+	} else {
+		updated.Status.Domain = domain
 	}
```

Comment:
The `else` block can be unwrapped:
```suggestion
	if !unique {
		log.Info("domain not unique, not setting status domain for IngressController", "namespace", ic.Namespace, "name", ic.Name)
This conversation was marked as resolved by danehans
 		return nil
	}

	updated.Status.Domain = domain
```